index.js

```diff
@@ -52,12 +52,26 @@ module.exports = function (options) {
       process.exit(1)
     }, 10000)
 
-    cluster.disconnect(function () {
-      restarting = false
-      clearTimeout(timeout)
-      logger.info('Cluster shutdown cleanly')
-      process.exit(0)
-    })
+    var workersAllExited= function() {
+      var runningWorker = _.find(workers, function (worker) {
+        return worker.state != 'exited'
+      })
+      return runningWorker === undefined
+    }
+
+    var cleanExit = function () {
+      if (workersAllExited()) {
+        restarting = false
+        clearTimeout(timeout)
+        logger.info('Cluster shutdown cleanly')
+        process.exit(0)
+        return
+      }
+
+      setTimeout(cleanExit, 250)
```

Comment:
Can we log a message here along the lines of "workers not yet exited, sleeping 250ms"?